sections/Chapter 6 Sprint 4/retrospective.tex

```diff
@@ -0,0 +1,32 @@
+
+We are waiting for validation from the project supervisor (report review).
+
+\subsection{Sprint goal}
+The goal of this sprint was to gather information about what to do next sprint, write missing report sections from the previous sprints and revise the report based on the feedback received from the supervisor. 
+
+\subsection{Reflection on the sprint}
+During previous sprints, focus had been dedicated to completing the implementation and deployment of the wordcount database and WordCount API. This resulted in parts of the report being unfinished, missing or in need of revision. Consequently, this meant that our focus this sprint had to be directed towards catching up on the report.
+During the third sprint a major contributing factor causing problems was identified, which was the length of the sprint. Previous sprints had been long, a total of four weeks, and resulted in a lack of overview and lack of flexibility. To accommodate this, we decided to reduce the sprint length significantly, down to two weeks. The benefit of this approach became evident as we were now able to better grasp the requirements of the sprint goal and appropriately granulated tasks to fulfill it. As the sprint was completed the resulting improvements could be observed clearly in the number of tasks completed and the number of story points that we achieved. 
+Another change was the addition of user stories to better define our tasks as it improved our understanding of the requirements within the sprint. This also made it easier for the product owner of the group to create tasks without having to consider specific implementation details and this also resulted in less duplication of tasks, which had been an issue previously.
+A final change that improved our focus and workflow during the sprint was the decision to postpone solving urgent issues for other groups, until a time that better suited us. During the beginning sprints, a lot of time had been dedicated to solving such issues.
+
+\subsection{What we learned}
+We learned that it is easier to keep an overview in smaller scale sprints as the resulting granulation of tasks is easier to do. When issues or new tasks were identified by our PO in previous sprints, we had more plans to change to accomodate new demands. With the smaller sprints changes to the plan are easier to handle. 
+We also learned that during previous sprints a lot of time had been spend creating tasks that were unnecessarily specific and the details of the implementation were often unclear at the time of creating them. We found that creating user stories and defining the tasks based on these user stories made the process easier. It also allowed the individual group member to have the freedom to decide how to fulfull the user story which resulted in a smoother workflow.  
```

Comment:
```suggestion
We also learned that, during previous sprints, a lot of time had been spent creating tasks that were unnecessarily specific, and the details of the implementation were often unclear at the time of creating them. We found that creating user stories and defining the tasks based on these made the process easier. It also allowed the individual group member to have the freedom to decide how to fulfill the user story which resulted in a smoother workflow.  
```

Reply:
"a lot of time had been spent creating tasks that were unnecessarily specific, and the details of the implementation were often unclear at the time of creating them."
Sounds like a contradiction to me - the tasks were very specific yet somehow the details of the implementation were unclear? As a reader with an outside perspective, this makes no sense.